prover/benches/logup_gkr.rs

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+use std::{marker::PhantomData, time::Duration, vec::Vec};
+
+use air::{
+    Air, AirContext, Assertion, AuxRandElements, ConstraintCompositionCoefficients,
+    EvaluationFrame, FieldExtension, LogUpGkrEvaluator, LogUpGkrOracle, ProofOptions, TraceInfo,
+    TransitionConstraintDegree,
+};
+use criterion::{criterion_group, criterion_main, BatchSize, BenchmarkId, Criterion};
+use crypto::MerkleTree;
+use math::StarkField;
+use winter_prover::{
+    crypto::{hashers::Blake3_256, DefaultRandomCoin},
+    math::{fields::f64::BaseElement, ExtensionOf, FieldElement},
+    matrix::ColMatrix,
+    DefaultConstraintEvaluator, DefaultTraceLde, Prover, StarkDomain, Trace, TracePolyTable,
+};
+
+const TRACE_LENS: [usize; 2] = [2_usize.pow(20), 2_usize.pow(21)];
+const AUX_TRACE_WIDTH: usize = 2;
+
+fn prove_with_lagrange_kernel(c: &mut Criterion) {
+    let mut group = c.benchmark_group("prove with Lagrange kernel column");
+    group.sample_size(10);
+    group.measurement_time(Duration::from_secs(20));
+
+    for &trace_len in TRACE_LENS.iter() {
+        group.bench_function(BenchmarkId::new("", trace_len), |b| {
+            let trace = LogUpGkrSimple::new(trace_len, AUX_TRACE_WIDTH);
+            let prover = LogUpGkrSimpleProver::new(AUX_TRACE_WIDTH);
+
+            b.iter_batched(
+                || trace.clone(),
+                |trace| prover.prove(trace).unwrap(),
+                BatchSize::SmallInput,
+            )
+        });
+    }
+}
+
+criterion_group!(lagrange_kernel_group, prove_with_lagrange_kernel);
+criterion_main!(lagrange_kernel_group);
+
+// LogUpGkrSimple
+// =================================================================================================
+
+#[derive(Clone, Debug)]
+struct LogUpGkrSimple {
```

Comment:
nit: seems like this is a mock execution trace, right? If so, maybe it should be called `LogUpGkrSimpleTrace`?

Reply:
Changed